front/lib/assistant/configuration_service.ts

```diff
@@ -0,0 +1,299 @@
+import type { Transaction } from "sequelize";
+
+import { DEFAULT_MCP_ACTION_DESCRIPTION } from "@app/lib/actions/constants";
+import type { ServerSideMCPServerConfigurationType } from "@app/lib/actions/mcp";
+import type { AgentActionConfigurationType } from "@app/lib/actions/types/agent";
+import {
+  createAgentActionConfiguration,
+  createAgentConfiguration,
+  unsafeHardDeleteAgentConfiguration,
+} from "@app/lib/api/assistant/configuration";
+import { getAgentConfigurationGroupIdsFromActions } from "@app/lib/api/assistant/permissions";
+import type { Authenticator } from "@app/lib/auth";
+import { AppResource } from "@app/lib/resources/app_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { ServerSideTracking } from "@app/lib/tracking/server";
+import type {
+  AgentConfigurationType,
+  PostOrPatchAgentConfigurationRequestBody,
+  Result,
+} from "@app/types";
+import { assertNever, Err, Ok } from "@app/types";
+
+/**
+ * Create Or Upgrade Agent Configuration If an agentConfigurationId is provided, it will create a
+ * new version of the agent configuration with the same agentConfigurationId. If no
+ * agentConfigurationId is provided, it will create a new agent configuration. In both cases, it
+ * will return the new agent configuration.
+ **/
+export async function createOrUpgradeAgentConfiguration({
+  auth,
+  assistant,
+  agentConfigurationId,
+  transaction,
+}: {
+  auth: Authenticator;
+  assistant: PostOrPatchAgentConfigurationRequestBody["assistant"];
```

Comment:
`PostOrPatchAgentConfigurationRequestBody` should not leak outside of the API endpoints so you likely want an actual type here? At the very least a clean alias.